core/sail/fts/lucene-api/src/main/java/org/eclipse/rdf4j/sail/lucene/LuceneSail.java

```diff
@@ -282,14 +282,20 @@ public SearchIndex getLuceneIndex() {
 	public NotifyingSailConnection getConnection()
 		throws SailException
 	{
-		return new LuceneSailConnection(super.getConnection(), luceneIndex, this);
+		if (!closed.get()) {
+			return new LuceneSailConnection(super.getConnection(), luceneIndex, this);
+		}
+		else {
+			throw new SailException("Sail is shut down or not initialized");
+		}
 	}
 
 	@Override
 	public void shutDown()
 		throws SailException
 	{
-		if(closed.compareAndSet(false, true)) {
+		logger.debug("LuceneSail shutdown");
```

Comment:
This call should be inside of the if statement so that it only ever occurs once.